mailpoet/lib/Automation/Integrations/MailPoet/Triggers/SegmentSubscribedTrigger.php

```diff
@@ -63,4 +71,29 @@ public function handleSubscription(SubscriberSegmentEntity $subscriberSegment):
       ],
     ]);
   }
+
+  public function isTriggeredBy(WorkflowRun $workflowRun): bool {
+    if ($workflowRun->getTriggerKey() !== $this->getKey()) {
+      return false;
+    }
+    $workflow = $this->workflowStorage->getWorkflow($workflowRun->getWorkflowId(), $workflowRun->getVersionId());
+    if (!$workflow) {
+      return false;
+    }
+
+    $triggerData = $workflow->getTrigger($workflowRun->getTriggerKey());
+    if (!$triggerData) {
+      return false;
+    }
+
+    $segmentSubject = $workflowRun->requireSingleSubject(SegmentSubject::class);
+    $segment = $segmentSubject->getSegment();
+    $stepArgs = $triggerData->getArgs();
+    if (!isset($stepArgs['segment_ids']) || !is_array($stepArgs['segment_ids'])) {
+      return false;
+    }
+    $segmentIds = $stepArgs['segment_ids'];
+    $anyList = 0;
+    return in_array($anyList, $segmentIds, true) || in_array($segment->getId(), $segmentIds, true);
+  }
```

Comment:
So I think this should be just `isTriggeredBy(array $args)`. The workflow, trigger, and run checks are all redundant, the `TriggerHandler` takes care of that. It still feels a bit weird to return to the trigger we just used to trigger the workflow but I see why it's needed.

(Ultimately, maybe the architecture is wrong. Maybe instead of calling `mailpoet_segment_subscribed` somewhere in our code and then landing in `SegmentSubscribedTrigger`, we should be always calling just `mailpoet_automation_trigger` with a trigger name and args. Then the trigger class would receive a workflow run and could implement the `run` method similarly to actions. I'll think about it, no need to change now.)

Reply:
If it simplifies things, `isTriggeredBy(array $args, array $subjects)` could also work, maybe.

Reply:
That is a good suggestion! Thank you. Addressed in [1a41901](https://github.com/mailpoet/mailpoet/pull/4371/commits/1a41901bf324435635221ea43f0fe917d678b69d)